client/src/main/scala/org/http4s/client/JavaNetClient.scala

```diff
@@ -0,0 +1,166 @@
+package org.http4s
+package client
+
+import cats.data.Kleisli
+import cats.effect.{Effect, Sync}
+import cats.implicits._
+import fs2.io.{readInputStream, writeOutputStream}
+import java.net.{HttpURLConnection, Proxy, URL}
+import javax.net.ssl.{HostnameVerifier, HttpsURLConnection, SSLSocketFactory}
+import org.http4s.internal.blocking
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.{Duration, FiniteDuration}
+
+/** A [[Client]] based on `java.net.HttpUrlConnection`.
+  *
+  * `JavaNetClient` adds no dependencies beyond `http4s-client`.  This
+  * client is generally not production grade, but convenient for
+  * exploration in a REPL.
+  *
+  * All I/O operations in this client are blocking.
+  *
+  * @param blockingExecutionContext An `ExecutionContext` on which
+  * blocking operations will be performed.
+  * @param ec The `ExecutionContext` to which work will be shifted
+  * back after blocking.
+  */
+sealed abstract class JavaNetClient private (
+  val connectTimeout: Duration,
+  val readTimeout: Duration,
+  val proxy: Option[Proxy],
+  val hostnameVerifier: Option[HostnameVerifier],
+  val sslSocketFactory: Option[SSLSocketFactory],
+  val blockingExecutionContext: ExecutionContext,
+)(implicit ec: ExecutionContext) {
```

Comment:
This is a good example of what I have in mind for the client and server configurations.  This can be evolved by adding more values.  Generating the `with` methods is tedious, but I think compatibility wins here.

Reply:
Maybe having types named `BlahClient` create a plain old `Client` is confusing.  We could all this a `JavaNetClientBuilder` or ` JavaNetClientConfig`, but I like the shape.